lib/diagnostic.rb

```diff
@@ -5,7 +5,7 @@ class Diagnostic
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    enum.max_by { |i| i.length }
```

Comment:
Yes, this will work. You can also use `.max` and pass it a custom comparator block:
```
enum.max { |a, b| a.length <=> b.length }
```